common/src/main/java/io/pravega/common/util/BlockingDrainingQueue.java

```diff
@@ -74,7 +85,23 @@ public BlockingDrainingQueue() {
             pending.cancel(true);
         }
 
-        return result != null ? result : Collections.emptyList();
+        return result != null ? result : new LinkedList<>();
```

Comment:
Collections.emptyList()

Reply:
This returns a `Queue`. `List` does not implement `Queue`.